From bfa60aa43077967a4ee89b1ce1796148830c39d1 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 27 Apr 2021 18:12:00 +0300
Subject: [PATCH] [SW-40350] fwupd: wrap all globals in struct

As a preparation for reworking the fwupd code to be busy-wait loopless,
move all the module globals inside state struct.
Later the module will obtain more global vars defining its state.

Change-Id: Ia3564bf83e689525027cd5224aae183bb7ee80ee
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index edf5508..ec68b55 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -30,9 +30,13 @@
 #define HL_FIT_LOAD_TYPE_PROP		"load-type"
 #define HL_FIT_PROTO_STATUS_PROP	"protocol-status"
 
-static bool is_fit_checked;
-static unsigned int map_size;
-static bool sending_error_flag;
+struct fwupd_state {
+	bool is_fit_checked;
+	unsigned int map_size;
+	bool sending_error_flag;
+};
+
+static struct fwupd_state state;
 
 void module_fwupd_setup(void)
 {
@@ -54,10 +58,10 @@
 	 * can't be change to a specific state therefore a flag was added
 	 * other option is to add an internal cmd level.
 	 */
-	if (sending_error_flag)
+	if (state.sending_error_flag)
 		return FOUND_SLOW_RETURN;
 
-	if (is_fit_checked)
+	if (state.is_fit_checked)
 		return FOUND_NO_RETURN;
 
 	cmd.val = regmap_readl(mmUPD_CMD);
@@ -132,7 +136,7 @@
 		goto fwupd_map_err;
 	}
 
-	map_size = size;
+	state.map_size = size;
 
 	if (cmd == HL_UPD_MAP_SRAM)
 		preboot_security_dynamic_img_sram_open();
@@ -185,8 +189,8 @@
 
 static int module_fwupd_load_type_run(void)
 {
-	is_fit_checked = !fit_load_images();
-	if (!is_fit_checked)
+	state.is_fit_checked = !fit_load_images();
+	if (!state.is_fit_checked)
 		return 1;
 
 	return 0;
@@ -231,7 +235,7 @@
 	err = module_fwupd_load_type();
 
 	if (cmd == HL_UPD_RUN_SPSRAM)
-		memset((void *)base_offset, 0, ROUND_UP4(map_size));
+		memset((void *)base_offset, 0, ROUND_UP4(state.map_size));
 
 	if (err) {
 		sts = HL_UPD_STS_VALID_ERR;
@@ -343,9 +347,9 @@
 
 	cmd.val = regmap_readl(mmUPD_CMD);
 
-	if (sending_error_flag) {
+	if (state.sending_error_flag) {
 		preboot_puts("FW UPD prepare ERROR_STATUS\n");
-		sending_error_flag = false;
+		state.sending_error_flag = false;
 	}
 
 	switch (cmd.cmd) {
